app/javascript/src/QuickStarts/components/CustomCatalog.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { useContext, useMemo } from 'react'
+import {
+  QuickStartCatalogHeader,
+  QuickStartCatalogSection,
+  QuickStartContext,
+  QuickStartTile,
+  getQuickStartStatus,
+  LoadingBox
+} from '@patternfly/quickstarts'
+import {
+  Gallery,
+  GalleryItem,
+  Text,
+  TextContent
+} from '@patternfly/react-core'
+
+import type {
+  QuickStart,
+  QuickStartContextValues } from '@patternfly/quickstarts'
+
+import './QuickStartContainer.scss'
+
+export const CustomCatalog: React.FC = () => {
+  const { activeQuickStartID, allQuickStartStates, allQuickStarts, loading } =
+    useContext<QuickStartContextValues>(QuickStartContext)
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- Following PF team guidelines
+  const CatalogWithSections = useMemo(
+    () => (
+      <>
+        <QuickStartCatalogSection>
```

Comment:
So here instead of repeating the same block 3 times I would do:
1. Define a map in a variable that would map the category "ID" (e.g. `basic-api-integration-setup`) to the title, e.g. "Basic API integration setup", and then just iterate over this map.